test/addons-napi/test_reference/test.js

```diff
@@ -11,80 +11,110 @@ const test_reference = require(`./build/${common.buildType}/test_reference`);
 // of a finalizer callback increments the finalizeCount property.
 assert.strictEqual(test_reference.finalizeCount, 0);
 
-{
-  // External value without a finalizer
-  let value = test_reference.createExternal();
-  assert.strictEqual(test_reference.finalizeCount, 0);
-  assert.strictEqual(typeof value, 'object');
-  test_reference.checkExternal(value);
-  value = null;
-  global.gc();
-  assert.strictEqual(test_reference.finalizeCount, 0);
+// Run each test function in sequence,
+// with an async delay and GC call between each.
+function runTests(i, title, tests) {
+  if (tests[i]) {
+    if (typeof tests[i] === 'string') {
+      title = tests[i];
+      runTests(i + 1, title, tests);
+    } else {
+      try {
```

Comment:
That could work, except `assert.doesNotThrow()` **loses the stack trace of the original exception**; it only displays the message. So I prefer not to use it except when checking for a very specific exceptional condition.